scripts/setup_offline_updates.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+usage() {
+    echo
+    echo "Usage: $0 <repo-type> <tar-file>"
+    echo
+    echo "Creates or updates an XCP-ng offline <repo-type> repository from the <tar-file> archive."
+    echo
+    echo "Arguments:"
+    echo "  repo-type   The repository type, based on its name on online mirrors. Example : 'updates'."
+    echo "  tar-file    The path to the archive containing the repository files."
+}
+
+if [ -z "$2" ]; then
+    echo "Error: argument(s) missing."
+    usage
+    exit 1
+fi
+
+set -eu
+
+REPO_TYPE="$1"
+TAR_FILE="$2"
+REPOS_DIR="/var/local/xcp-ng-repos"
+REPONAME="xcp-ng-offline-$REPO_TYPE"
+
+if [ ! -e "$TAR_FILE" ]; then
+    echo "Error: file $TAR_FILE not found."
+    exit 2
+fi
+
+if ! tar -tf "$TAR_FILE" | grep -q "^${REPO_TYPE}/$"; then
```

Comment:
truncated/corrupted tar file detection needs `set -o pipefail`

Reply:
Well, the story with `pipefail` is funny. `grep -q` stops early as soon as it finds a match, which sends SIGPIPE to tar, thus we end with return code 141 and the whole condition with `!` never evaluates to false.

If the tar file is corrupted, we'll have an error when we extract it anyway, right?

Reply:
> If the tar file is corrupted, we'll have an error when we extract it anyway, right?

Yes, but we'll have removed the old repo already

Reply:
I'll test the tarball integrity first then.